Floofbot/Services/UserRoleRetentionService.cs

```diff
@@ -72,7 +66,7 @@ public async Task RestoreUserRoles(IGuildUser user)
                 // restore each role 1 by 1
                 foreach (string roleId in oldUserRolesList)
                 {
-                    SocketRole role = GetRole(roleId, user.Guild);
+                    IRole role = user.Guild.GetRole(Convert.ToUInt64(roleId));
 
                     if (role == null) // role does not exist
```

Comment:
Sorry, probably should've left a comment on this line but I was also requesting a debug log here as well.